### PR TITLE
Implement sealed class Lenses

### PR DESCRIPTION
**Pull request by maksym-moroz**
Addresses #2829 
Implement and enable Lens generation for sealed class properties when these properties are abstract and without extension receiver, subclasses are data classes that override properties in the constructor. 
If that's not the case emit an info log message explaining why Lens is not being generated.
Additional cleanup to target logic.

All tests pass

### CODE REVIEW

**Comment by serras** [comment]
@maksym-moroz can you run the `spotlessApply` task and re-commit?

**Comment by maksym-moroz** [comment]
@serras I did just that and it didn't show any issues with build

**Comment by serras** [comment]
The automatic Spotless commit was not working, I've done it manually. This should (hopefully) turn the build green.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip
```
Comment (serras) [inline gradle/wrapper/gradle-wrapper.properties:3]: What is the reasoning for this change?

File: arrow-libs/optics/arrow-optics-ksp-plugin/src/test/kotlin/arrow/optics/plugin/LensTests.kt [hunk 2]
```diff
@@ -130,4 +130,23 @@ class LensTests {
       |}
       """.compilationSucceeds()
   }
+
+  @Test
+  fun `Lens for sealed class property`() {
```
Comment (serras) [inline arrow-libs/optics/arrow-optics-ksp-plugin/src/test/kotlin/arrow/optics/plugin/LensTests.kt:135]: I am missing three other cases for test:
1. More than one class extending the sealed one,
2. Data classes defined outside of the sealed class,
3. Sealed class with zero inheritors.

**Review by serras — CHANGES_REQUESTED** [review]
Thanks for this great work. Not only implementing the feature, but also cleaning the code a bit on the go :)

I've left a few comments about things (mostly tests) that I think should be in the code before merging it in `main`.

**Pull request merged**